example/eblob_backend.c

```diff
@@ -872,13 +874,156 @@ int blob_defrag_start(void *priv, enum dnet_backend_defrag_level level)
+	err = dnet_send_reply_threshold(state, cmd, re, sizeof(struct dnet_iterator_response), more);
+
+	/* we have to convert response back, since it can be reused, for example like error response */
+	dnet_convert_iterator_response(re);
+	return err;
+}
+
+static int blob_send(struct eblob_backend_config *cfg, void *state, struct dnet_cmd *cmd, void *data)
+{
+	struct eblob_backend *b = cfg->eblob;
+	struct dnet_server_send_request *req = data;
+	struct dnet_raw_id *ids;
+	struct dnet_iterator_response re;
+	struct dnet_server_send_ctl *ctl;
+	int *groups;
+	int i, err;
+
+	struct dnet_ext_list elist;
+	static const size_t ehdr_size = sizeof(struct dnet_ext_list_hdr);
+	struct eblob_key key;
+	struct eblob_write_control wc;
+	uint64_t data_offset, record_offset;
+
+	dnet_ext_list_init(&elist);
+
+
+	/* structure has been already checked and has been proved to be correct no need to perform sanity checks */
+	dnet_convert_server_send_request(req);
+
+	ids = (struct dnet_raw_id *)(req + 1);
+	groups = (int *)(ids + req->id_num);
+
+	memset(&re, 0, sizeof(struct dnet_iterator_response));
+	re.total_keys = req->id_num;
+
+	/*
+	 * Set NEED_ACK bit to signal server-send controller that we want
+	 * to send final ACK when controller will be destroyed, which in turn
+	 * will happend after all WRITE commands are completed.
+	 *
+	 * Command will be copied internally in @dnet_server_send_alloc(),
+	 * thus it is safe to clear that bit afterward.
+	 */
+	cmd->flags |= DNET_FLAGS_NEED_ACK;
+
+	ctl = dnet_server_send_alloc(state, cmd, req->iflags, groups, req->group_num);
+	if (!ctl) {
+		err = -ENOMEM;
+		goto err_out_exit;
+	}
+
+	/*
+	 * Deliberately clear NEED_ACK bit
+	 * This function will iterate over provided ids,
+	 * read them from the blob and queue WRITE command to remote groups.
+	 * When single WRITE command completes, it will send response back to client.
+	 *
+	 * If we send ACK in the middle, it will force client to stop accepting further responses.
+	 *
+	 * If there will be an error, it will be sent to client too.
+	 * If there is an error with command processing (like fail to allocate memory),
+	 * iteration will stop and error will be returned from this function to the higher layer,
+	 * which in turn will force ACK message to client with error code.
+	 */
+	cmd->flags &= ~DNET_FLAGS_NEED_ACK;
+
+
+
+	for (i = 0; i < req->id_num; ++i) {
+		memcpy(key.id, ids[i].id, EBLOB_ID_SIZE);
+
+		err = blob_lookup(b, &key, &wc);
+		if (err < 0) {
+			dnet_backend_log(cfg->blog, DNET_LOG_ERROR, "%s: EBLOB: blob_send: lookup: %d: %s",
+					 dnet_dump_id_str(key.id), err, strerror(-err));
+			goto err_out_send_fail_reply;
+		}
+
+		re.key = ids[i];
+		re.flags = wc.flags; // these flags corresponse to DNET_RECORD_FLAGS_*
+		re.status = 0;
+		re.iterated_keys = i;
+		re.size = wc.total_data_size;
+		// set iterator response id to differentiate various commands
+		// client can use cmd->backend_id from reply though
+		re.id = cmd->backend_id;
```

Comment:
No, it can't. It behaves quite differently from how iterator runs, it is more like bulk IO request, the only common thing is response structure sent to client.